### ISSUE TITLE
Phase 2: clustered planning

### ISSUE THREAD

**Issue by piskuliche** [piskuliche/RBFENetworkMap#21]
Part of #18. Branch `feat/phase2-clustered-planning` off `epic/network-knobs`, PR back into it.

> **Note on prior art.** #8 / #10 proposed a core-based clustering design that was closed as
> not planned, and that branch is deliberately **not** being revived. This is a different
> approach. Do not pull from `feature/core-cluster-networks`.

HiMap's cost result: `Σᵢ nᵢ·ln(nᵢ) < n·ln(n)`, so `γ_min = γ_edge · n · ln(n/d)` for `d` balanced
clusters. n=100 in 5 clusters: ~460 → ~190 edges, a 59% saving at maintained precision;
worst-case imbalance still saves 30–50%.

This composes with the CBFE machinery rather than duplicating it — inter-cluster edges are
exactly where a counterpoised edge belongs.

## 2.1 Clusterers — new `src/rbfenetmap/core/clustering.py`

```python
cluster_by_charge(ligands)      -> dict[str, int]
cluster_by_scaffold(ligands)    -> dict[str, int]   # RDKit Murcko / rdScaffoldNetwork
cluster_by_fingerprint(ligands, *, n_clusters=None, cutoff=None) -> dict[str, int]
```

**Use `scipy.cluster.hierarchy.linkage` + `fcluster` on the condensed 1−Tanimoto matrix, not
sklearn.** Konnektor uses HDBSCAN/KMeans and HiMap uses DBSCAN, but scipy is already a core
dependency and sklearn is not; average-linkage on Tanimoto distance is the standard
chemoinformatics choice and needs nothing new. Reuse `fingerprint_pair_similarities` from
`core/pairs.py`.

## 2.2 A knob, not a planner

Add to `NetworkOptions`: `cluster_by: {none, charge, scaffold, fingerprint} = "none"` and
`cluster_bridges: int = 2`. A separate `ClusteredPlanner` would not compose — users want
clustering *and* cycle coverage *and* CBFE bridges together, which is what
`MSTRedundancyPlanner` already orchestrates.

## 2.3 The enabling refactor

`core/cbfe.py:select_cbfe_bridges(graph, ligands, pool)` derives its partition internally from
`nx.connected_components`. Extract the partition as a parameter:

```python
select_bridges(partition: Mapping[str, int], ligands, pool, *, n_per_pair: int = 1)
```

`select_cbfe_bridges` becomes a thin caller passing the connected-components partition.
Clustered planning passes the clusterer's partition with `n_per_pair=cluster_bridges`.
`bridge_rank_key` and `component_centrality` are reused unchanged.

**`cluster_bridges=2` is the deliberate default**: two bridges between a cluster pair put the
bridge itself on a cycle, applying the FEP+ every-edge-in-a-cycle invariant precisely to the
riskiest edges in the network. Konnektor's `MstConcatenator` uses the same default for the same
reason.

`diagnose` reports `Σ nᵢ ln(nᵢ)` against `n ln n` so the saving is visible.

## Acceptance

- [ ] Golden baseline still byte-identical on the default path and under `--compat v0.4`
- [ ] Charge clusterer on mixed-charge ligands yields exactly the charge classes
- [ ] `cluster_bridges=2` puts every inter-cluster edge on a cycle
- [ ] The two CBFE invariants survive the `select_bridges` refactor: degree padding never spends
      a CBFE edge; cycle closure prefers RBFE even when dearer
- [ ] No new third-party dependency
